src/sentry/web/frontend/base.py

```diff
@@ -419,11 +434,12 @@ def needs_sso(self, request: Request, organization):
         return False
 
     def convert_args(self, request: Request, organization_slug=None, *args, **kwargs):
-        active_organization = self.get_active_organization(
-            request=request, organization_slug=organization_slug
-        )
+        organization: Optional[Organization] = None
+        if self.active_organization:
+            for org in Organization.objects.filter(id=self.active_organization.id):
+                organization = org
 
-        kwargs["organization"] = active_organization
+        kwargs["organization"] = organization
```

Comment:
Region endpoints really do need the full organization object.  I think the next refactoring looks like splitting the View subclasses by wether they will end up in the region or control silo so that the typing and handling of this argument is consistent and documented.